geocat/comp/climatologies.py

```diff
@@ -518,24 +561,54 @@ def climatology_average(
     # Retrieve calendar name
     calendar = _infer_calendar_name(dset[time_dim])
 
+    attrs = {}
+    if keep_attrs or keep_attrs is None:
+        attrs = dset.attrs
+
     if freq == 'season':
-        attrs = {}
-        if keep_attrs or keep_attrs is None:
-            attrs = dset.attrs
-        if xr.infer_freq(dset[time_dim]) != 'MS':
+
+        if custom_seasons:
+            if type(custom_seasons) == str:
+                custom_seasons = [custom_seasons]
+            valid_season_check = [s in seasons_dict for s in custom_seasons]
+            if not all(valid_season_check):
+                bad_seasons = [
+                    custom_seasons[i]
+                    for i, x in enumerate(valid_season_check)
+                    if not x
+                ]
+                raise ValueError(
+                    f"contributed: climatology_average: bad season(s): {bad_seasons}. Valid seasons include: {list(seasons_dict.keys())}"
+                )
```

Comment:
As you saw this morning, I think we can improve the readability on this particular section. 

```suggestion
            invalid_seasons = [s for s in custom_seasons if s not in seasons_dict]
            if invalid_seasons:
                raise ValueError(
                    f"contributed: climatology_average: bad season(s): {invalid_seasons}. Valid seasons include: {list(seasons_dict.keys())}"
                )
```

Reply:
This is much better, thanks